roles/pool/src/lib/mod.rs

```diff
@@ -2,3 +2,107 @@ pub mod error;
 pub mod mining_pool;
 pub mod status;
 pub mod template_receiver;
+
+use async_channel::{bounded, unbounded};
+
+use mining_pool::{get_coinbase_output, Configuration, Pool};
+use template_receiver::TemplateRx;
+use tracing::{error, info, warn};
+
+use tokio::select;
+
+pub struct PoolSv2;
+
+impl PoolSv2 {
+    pub async fn start(config: Configuration) {
+        let (status_tx, status_rx) = unbounded();
+        let (s_new_t, r_new_t) = bounded(10);
+        let (s_prev_hash, r_prev_hash) = bounded(10);
+        let (s_solution, r_solution) = bounded(10);
+        let (s_message_recv_signal, r_message_recv_signal) = bounded(10);
+        let coinbase_output_result = get_coinbase_output(&config);
+        let coinbase_output_len = match coinbase_output_result {
+            Ok(coinbase_output) => coinbase_output.len() as u32,
+            Err(err) => {
+                error!("Failed to get Coinbase output: {:?}", err);
+                return;
+            }
+        };
+        let tp_authority_public_key = config.tp_authority_public_key;
+        let template_rx_res = TemplateRx::connect(
+            config.tp_address.parse().unwrap(),
+            s_new_t,
+            s_prev_hash,
+            r_solution,
+            r_message_recv_signal,
+            status::Sender::Upstream(status_tx.clone()),
+            coinbase_output_len,
+            tp_authority_public_key,
+        )
+        .await;
+
+        if let Err(e) = template_rx_res {
+            error!("Could not connect to Template Provider: {}", e);
+            return;
+        }
+
+        let pool = Pool::start(
+            config.clone(),
+            r_new_t,
+            r_prev_hash,
+            s_solution,
+            s_message_recv_signal,
+            status::Sender::DownstreamListener(status_tx),
+        );
+
+        // Start the error handling loop
+        // See `./status.rs` and `utils/error_handling` for information on how this operates
+        tokio::spawn(async move {
```

Comment:
I tried to manually launch the pool executable, and all I got was this:
```
    Finished dev [optimized + debuginfo] target(s) in 7.52s
     Running `/home/plebhash/develop/stratum/roles/target/debug/pool_sv2 -c config-examples/pool-config-local-tp-example.toml`
2024-08-07T00:32:11.955091Z  INFO pool_sv2::lib::template_receiver: Connected to template distribution server at 127.0.0.1:8442
2024-08-07T00:32:12.257651Z  INFO roles_logic_sv2::handlers::common: Received SetupConnectionSuccess: version=2, flags=0
2024-08-07T00:32:12.257799Z  INFO pool_sv2::lib::mining_pool: PUB KEY: Ok([TxOut { value: 0, script_pubkey: Script(OP_0 OP_PUSHBYTES_20 ebe1b7dcc293ccaa0ee743a86f89df8258c208fc) }])
2024-08-07T00:32:12.257857Z  INFO pool_sv2::lib::mining_pool: Starting up pool listener
2024-08-07T00:32:12.258048Z  INFO pool_sv2::lib::mining_pool: Listening for encrypted connection on: 0.0.0.0:34254
➜  pool git:(2024-07-15-roles-testing) ✗ 
```

the executable would simply exit, it would not listen for connections.

I tried removing this `tokio:spawn`, leaving the `loop` in the same thread... that got the binary to listen for connections again, but then running `cargo test` on the `tests-integration` made it hang indefinitely

Reply:
overall the lesson here is that we need to make sure we're not breaking the basic behavior of all the refactored roles

Reply:
overall the lesson here is that we need to make sure we're not breaking the basic behavior of all the refactored roles